cpp/inc/bond/core/transforms.h

```diff
@@ -377,7 +377,7 @@ class RequiredFieldValiadator
     {}
 
 private:
-    void MissingFieldException() const;
+    BOND_NORETURN void MissingFieldException() const;
```

Comment:
Looks like Clang, GCC, and MSVC only need `[[noreturn]]` or `__declspec(noreturn)` or `__attribute__((noreturn))` on the declaration. I was confusing these with SAL annotations, which get weird if the declaration and definition are different.